src/languageservice/parser/jsonParser07.ts

```diff
@@ -646,6 +647,16 @@ function validate(
       return node.type === type || (type === 'integer' && node.type === 'number' && node.isInteger);
     }
 
+    function matchesSchemaType(schema: JSONSchema): boolean {
+      /* when schema type is object and it contains oneOf then needs to validate that particular types
+        reference issue 692
+      */
+      if (schema.type === 'object' && typeof schema.additionalProperties === 'object' && schema.additionalProperties.oneOf) {
+        return true;
+      }
+      return node.type === schema.type || (schema.type === 'integer' && node.type === 'number' && node.isInteger);
```

Comment:
Shouldn't this just call `matchesType` from above? It looks like it is duplicating the code.